D3521R0.md

```diff
@@ -200,80 +200,46 @@ Given an example like:
 Conceptually: `try_cast<@*type*@>(@*subject*@)` is invoked, which returns an
 optional result that can be tested and dereferenced to match the *subpattern*.
 
-More precisely: Let the result of `try_cast<@*type*@>(@*subject*@)` be `__temp`
-where `try_cast` is looked up by ADL-only (same as `get` for structured bindings).
-The pattern matches if `__temp` contextually converts to `true` and
+More precisely: Let `E` be
+`std::try_cast<std::remove_cv_t<@*type*@>>(@*subject*@)`.  The match succeeds
+if and only if: `E` is well-formed; `E` is implicitly convertible to
```

Comment:
`*E` is convertible to `type` right? Also I think would want stronger, like.. actually _is_? Would be weird if `try_cast<string>` could return `char const**`

Reply:
I was thinking of the case where your matcher had `const int: ` in it, and the `try_cast()` evaluates to `int *`.  That's fine, but I agree this allows for too much.